app/lib/helpers/job_helper.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Helpers
+  module JobHelper
+    class << self
+      def correct_feature?(job_invocation, feature)
+        job_invocation.job_category == ::ForemanLeapp::JOB_CATEGORY &&
+          RemoteExecutionFeature.find_by(job_template_id: job_invocation.pattern_template_invocations
+                                                                        .first
+                                                                        &.template_id)
```

Comment:
It is not needed to load the entire (but possibly `nil`) `pattern_template_invocation` and then just take a field out of it. Could you `pluck` and then `first`?